### ISSUE TITLE
Idea: More damage types / custom damage types

### ISSUE THREAD

**Issue by James103** [yeelp/Distinct-Damage-Descriptions#2]
Currently, you only have the mod setup for three different damage types: Slashing, Piercing, and Bludgeoning. Do you have plans to add more, or even allow packmakers to add in their own custom damage types?

**Comment by yeelp** [comment]
In short/TL;DR:

 **additional damage types**: potentially, it depends on the state of the mod after physical damage types are done for both ranged and melee attacks, and would mostly stick to D&D's system for damage types, and be only for elemental attacks.

**user defined damage types**: no.

As for a more detailed explanation, the question you've asked becomes a little more subtle, as this is now entering the realm of physical versus ranged versus spell attacks. 

As for physical attacks, no more will be added. The three I have should be sufficient. Swords enchanted with Fire Aspect would inflict slashing on a hit and set the target ablaze, which does normal Minecraft fire damage, which bypasses armor (which is different from standing *in* a fire, which *is* blocked by armor).

As for ranged attacks, the answer is also no. Arrows, and other material projectiles can inflict one of the three physical damage types on impact. Enchanted arrows with Flame would inflict piercing on hit and set the target ablaze, see description above.

Minecraft doesn't have spells, but other mods can add them, and Minecraft also has fire damage itself, so some elemental damage types aren't out of the question, and have been considered. For the most part, if I were to add additional damage types, I would stick to D&D's damage types, with either some extra additions (maybe void damage, although that bypasses armor anyway) or removals (thunder damage can be removed and replaced with sonic damage for explosions perhaps).

User defined damage types would probably make configs a huge mess that'd be hard to understand, so I probably won't do that. It would also make understanding the combat of a modpack far harder than needed, as users now need to figure out what damage types exist, what weapons do what damage types, what mobs do what damage types, what mobs are resistant/immune to what damage types and so on (as this is already all configurable). The price of entry becomes way too high in my opinion, regardless of how cool the idea of custom damage types are.

**Comment by yeelp** [comment]
Although potentially limited in application, consider commit b6c6277 a change in heart. Found a good way to implement them using JSON with custom creature types.
